core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -74,6 +76,43 @@ static __thread struct argv_environ_values *argv_environ;
 static __thread struct addr_pool *addr_pool;
 #endif
 
+#define NANOSECONDS_PER_SECOND 1000000000
+
+
+static uint64
+calculate_monotonic_clock_frequency()()
+{
+    LARGE_INTEGER frequency;
+    QueryPerformanceFrequency(&frequency);
+    return (uint64)frequency.QuadPart;
+
+}
+
+static __wasi_timestamp_t
+from_file_time_to_wasi_timestamp(FILETIME Filetime)
+{
+    __wasi_timestamp_t *attempt;
+        
+    static const uint64 NTToUnixEpoch =
```

Comment:
It'd be good to explain (or add a reference to a resource) where does the value come from.